src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> handlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlerMap) {
+        this.handlerMap = handlerMap;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return handlerMap.get(operation);
```

This line could cause a `NullPointerException` later in the code if no handler is found in the map for the given `operation`. It's safer to check if the result is `null` and throw a more specific exception, like `UnsupportedOperationException`, to fail fast and provide a clearer error message.